internal/scyllaCloudSDK/endpoints.go

```diff
@@ -0,0 +1,79 @@
+package scyllaCloudSDK
+
+import (
+	"errors"
+	"fmt"
+)
+
+func (c *Client) ListCloudProviders() ([]CloudProvider, error) {
+	var result []CloudProvider
+	if err := c.get("/deployment/provider", &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (c *Client) ListCloudProviderRegions(providerId int64) ([]CloudProviderRegion, error) {
+	var result []CloudProviderRegion
+	path := fmt.Sprintf("/deployment/provider/%d/region", providerId)
+	if err := c.get(path, &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (c *Client) ListClusters() ([]Cluster, error) {
+	type Item struct {
+		Value Cluster     `json:"Value"`
+		Error interface{} `json:"Error"`
+	}
+	var result []Item
+	path := fmt.Sprintf("/account/%d/cluster", c.accountId)
+	if err := c.get(path, &result); err != nil {
+		return nil, err
+	}
+
+	clusters := make([]Cluster, len(result))
+	for i, item := range result {
+		if item.Error != nil {
+			return nil, errors.New(fmt.Sprintf("cluster error: %v", item.Error))
```

Comment:
errors.Errorf

Reply:
it would be good practice to include in err message which cluster (is item.Value empty?) does the error belong to.

are you positive is item[x].Error is not-nil then the whole method should return non-nil error? doesn't non-nil item.Error mean the cluster is in error state (e.g. bootstrap failed) thus it should not fail the ListClusters() method?

Reply:
> it would be good practice to include in err message which cluster (is item.Value empty?) does the error belong to.
> 
> are you positive is item[x].Error is not-nil then the whole method should return non-nil error? 

no, probably not, but at the time of writing this it was unclear to me how errors here should be handled; I'll just proxy the return values along with errors from this method and handle errors downstream

> doesn't non-nil item.Error mean the cluster is in error state (e.g. bootstrap failed) 

no idea really, the attached error is the error returned from https://github.com/scylladb/siren/blob/f384b903f57f67f593fa58d58f4054bbbc3f07d9/cluster/cluster.go#L556 for each of the clusters, I didn't go into more detail

Reply:
Discussed on slack - we're leaving a TODO here for now as it's not clear how to handle it. The API design is weird.